### PR TITLE
Compile paint code as cpp.

### PR DESCRIPTION
**Pull request by ZehMatt**
In order to make it easier to refactor paint related code I renamed paint.c to paint.cpp to be compiled as c++ code. I had to fix a few minor things but nothing crucial. I also applied our current code style guidelines.

### CODE REVIEW

File: src/openrct2/paint/paint.cpp [hunk 1]
```diff
@@ -65,6 +70,7 @@ static void paint_ps_image_with_bounding_boxes(rct_drawpixelinfo * dpi, paint_st
 static void paint_ps_image(rct_drawpixelinfo * dpi, paint_struct * ps, uint32 imageId, sint16 x, sint16 y);
 static uint32 paint_ps_colourify_image(uint32 imageId, uint8 spriteType, uint32 viewFlags);
 
+extern "C"
```
Comment (Gymnasiast) [inline src/openrct2/paint/paint.cpp]: extern "C" methods go together in a block.

**Comment by Gymnasiast** [comment]
- C++ filenames are in TitleCase. It should therefore be named Paint.cpp
- C++ functions are in TitleCase. N.B.: the function names in the extern "C" block should be in snake_case, like all other C functions.
- The extern "C" block in a C++ file should only act as a "proxy", that is: all of the actual logic should be in C++, with the extern "C" functions calling the C++ ones.

**Comment by ZehMatt** [comment]
I can rename the file however I would do more refactor work in another PR, I want to refactor the sorting and probably the overall drawing code. Would it be ok if I just rename the file to be uppercase for now. The next PR is almost ready.

**Comment by Gymnasiast** [comment]
@ZehMatt Well, alright, rename the file and I'll merge it if there is nothing else.

**Pull request merged**